ql/src/java/org/apache/hadoop/hive/ql/processors/ShowProcesslistProcessor.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.processors;
+
+import com.google.common.base.Joiner;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Schema;
+import org.apache.hadoop.hive.ql.session.ProcessListInfo;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_NULL_FORMAT;
+import static org.apache.hadoop.hive.serde2.MetadataTypedColumnsetSerDe.defaultNullString;
+
+/**
+ * List operations/queries being performed in sessions within hiveserver2
+ */
+public class ShowProcesslistProcessor implements CommandProcessor{
+  private static final Logger LOG = LoggerFactory.getLogger(ShowProcesslistProcessor.class.getName());
+  private static final SessionState.LogHelper console = new SessionState.LogHelper(LOG);
+  private List<ProcessListInfo> liveQueries = null;
+
+  public void setup(List<ProcessListInfo> liveQueries) {
+    this.liveQueries = liveQueries;
+  }
+
+  /**
+   * Creates a Schema object with running operation details
+   * @return
+   */
+  private Schema getSchema(){
+    Schema sch = new Schema();
+    sch.addToFieldSchemas(new FieldSchema("User Name", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Ip Addr", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Execution Engine", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Session Id", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Session Active Time (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Session Idle Time (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Query ID", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("State", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Opened Timestamp (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Elapsed Time (s)", "string", ""));
+    sch.addToFieldSchemas(new FieldSchema("Runtime (s)", "string", ""));
```

Comment:
what Runtime stands for - is it just execution time. would Elapsed Time show full query time: queuing + compilation + waiting + execution?

Reply:
Runtime shows only the operation's execution time, while the Elapsed time covers the total time since the operation was initialized to Closed. If the operation is still running, it would be the time between the current time and the operation's initialized time.